server/handlers/product.js

```diff
@@ -0,0 +1,45 @@
+var dao = require("../dao/dao");
+const { Product } = require("../models/product");
+const {
+  searchStringValidator,
+  productCategoryValidator,
+  idsValidator,
+} = require("./shared_validators");
+const { validationResult } = require("express-validator");
+
+exports.getProductsByIDValidatorChain = [
+  searchStringValidator,
+  productCategoryValidator,
+  idsValidator,
+];
+
+exports.getProductsByIDHandler = async function (req, res, next) {
+  const errors = validationResult(req);
```

Comment:
This is now embedded in the checkValidationErrorMiddleware. Use the middleware and remove this code.